pkg/runner/cluster_k8s.go

```diff
@@ -221,9 +248,30 @@ func (*ClusterK8sRunner) Run(ctx context.Context, input *api.RunInput, ow io.Wri
 
 				podName := fmt.Sprintf("%s-%s-%d", jobName, g.ID, i)
 
-				logs := getPodLogs(client, podName)
+				// Log monitor.
+				path := filepath.Join(
+					runDir,
+					podName,
+				)
+				err = os.MkdirAll(path, 0777)
+				if err != nil {
+					return err
+				}
 
-				fmt.Print(logs)
+				f, err := os.Create(filepath.Join(path, "stdoutanderr.json"))
```

Comment:
I am not aware of a way to split `stdout` and `stderr` on k8s - I think there is no such feature.

Reply:
We could just let the sidecar do this.

Reply:
But can we not talk to docker directly?

Reply:
`sidecar` is on the remote cluster. This is on the `daemon` or on the `client` side, which currently is on the devs machine. I don't see any benefit in the logs going from individual containers to the sidecar container and then somehow again to the testground daemon and client.

What is the benefit of `sidecar` managing `logs` at all? This only introduces complexity.

Reply:
> But can we not talk to docker directly?

Not sure why we want to go one-level deeper and avoid using the Kubernetes APIs, when this is how we manage to workloads - we schedule `pods` with the Kubernetes API, we monitor their state with the Kubernetes API, why would we have to investigate what Kubernetes is doing under the hood and attach to individual containers for logs?

Reply:
My point was that the sidecar talks directly to the docker daemons on each of these nodes and _can_ distinguish between STDOUT and STDERR. Really, we don't want to send this back to the daemon anyways. If we're just dropping stderr/stdout to a directory in S3, the sidecar code can (could?) already do this.

However, if we're going going to start writing logs to a file, we should be able to just collect stdout (should only include stats) with the daemon and sidestep this.

Reply:
@Stebalien if we talk to the k8s API, then after a test is detected to be complete, we can get the logs, and only then remove the pods. If we talk directly to the docker daemon from the `sidecar`, I am not sure how we can synchronise the fact that testground will be removing the pods after the test plan completes - how do we get guarantee that all logs have been streamed to the sidecar from all instances?

That synchronisation is easier on Kubernetes, because we control the testplan flow there.

Reply:
Hm. _That's_ a good point. Given that, I'd really consider redirecting stderr to a file within the container.